strawberry/sanic/views.py

```diff
@@ -0,0 +1,96 @@
+import json
+
+from sanic.exceptions import abort
+from sanic.response import HTTPResponse, html
+from sanic.views import HTTPMethodView
+from strawberry.http import GraphQLHTTPResponse, process_result
+from strawberry.types import ExecutionResult
+
+from ..schema import BaseSchema
+from .graphiql import render_graphiql_page
+
+
+class GraphQLView(HTTPMethodView):
+    """
+    Class based view to handle GraphQL HTTP Requests
+
+    Args:
+        schema: strawberry.Schema
+        graphiql: bool, default is True
+
+    Returns:
+        None
+
+    Example:
+        app.add_route(
+            GraphQLView.as_view(schema=schema, graphiql=True),
+            "/graphql"
+        )
+    """
+
+    methods = ["GET", "POST"]
+
+    def __init__(self, schema: BaseSchema, graphiql: bool = True):
+        self.graphiql = graphiql
+        self.schema = schema
+
+    def get_root_value(self):
+        return None
+
+    def get_context(self, request):
+        return {"request": request}
```

Comment:
can we return a class like we did here? https://github.com/strawberry-graphql/strawberry/pull/723

I think it's better to have a dataclass for this :)

Reply:
Implemented in  commit d1e0ced